main.py

```diff
@@ -25,6 +25,17 @@
 
 API_URL="https://api.github.com"
 
+def test_rate():
+    url = "https://api.github.com/rate_limit"
+    response = requests.get(url)
+    json = response.json()
+    remaining = json['resources']['core']['remaining']
+    return remaining
+
+if test_rate() == 0:
```

Comment:
This feels a bit aggressive; can't we just check if we got an error, and then report it?
Why would we have to explicitly check for errors independently of the actual request?

I feel this adds a https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use error.

Reply:
I just felt it was cleaner. Idk though.